sys/net/application_layer/dns/dns.c

```diff
@@ -129,6 +131,13 @@ static int _parse_dns_reply(uint8_t *buf, size_t len, void* addr_out, int family
             bufpos += addrlen;
```

Comment:
This pointer addition still needs a check, because `addrlen` is attacker controlled. This could be used for DOS by overflowing the pointer to point into invalid memory, which would not be detected by any of the bounds checks you implemented.

Reply:
But `bufpos` is immediately checked after the continue within `_skip_hostname()`.

Reply:
But I can make `bufpos` smaller than the end of the buffer by overflowing it...

Reply:
Fixed.